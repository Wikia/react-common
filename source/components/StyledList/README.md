Default:
```js
<StyledList>
    <li>First one</li>
    <li>Second one</li>
    <li>Third one</li>
</StyledList>
```
A list with "big items". This makes padding borader and the font size bigger in each element

```js
<StyledList bigItems>
    <li>First one</li>
    <li>Second one</li>
    <li>Third one</li>
</StyledList>
```

A list with bolded elements.

```js
<StyledList boldItems>
    <li>First one</li>
    <li>Second one</li>
    <li>Third one</li>
</StyledList>
```
A list with a separator line between each element.

```js
<StyledList linesBetween>
    <li>First one</li>
    <li>Second one</li>
    <li>Third one</li>
</StyledList>
```
A list with links. This is necessary to ensure the elements are fully clickable and are without any surrounding gaps.

```js
<StyledList isLinked>
    <li><a href="http://wikia.com">Linked element 1</a></li>
    <li><a href="http://wikia.com">Linked element 2</a></li>
    <li><a href="http://wikia.com">Linked element 3</a></li>
</StyledList>
```
A list with ellipsis.
```js
<StyledList hasEllipsis>
    <li>First one...</li>
    <li>Second one...</li>
    <li>Third one...</li>
</StyledList>
```
