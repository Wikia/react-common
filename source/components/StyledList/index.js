import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { ellipsis } from '../../utils/css';

export const LIST_VERTICAL_PADDING = 11;

const StyledList = styled.ul`
    list-style-type: none;
    margin: 0;
    padding: 7px 18px;

    ${({ bigItems }) => bigItems && css`
        padding-bottom: 2px;
        padding-top: 2px;
    `}

    > li {
        font-size: ${({ theme }) => theme.font_size.s};
        font-weight: normal;
        line-height: 1em;
        padding: ${LIST_VERTICAL_PADDING}px 0;

        ${({ bigItems }) => bigItems && css`
            font-size: ${({ theme }) => theme.font_size.base};
            padding-bottom: 16px;
            padding-top: 16px;
        `}

        ${({ boldItems }) => boldItems && css`
            font-weight: bold;
        `}

        ${({ linesBetween, theme }) => linesBetween && css`
            border-bottom: solid 1px ${theme.color.fandom_light_gray};

            &:last-child {
                border: 0;
            }
        `}

        ${({ hasEllipsis }) => hasEllipsis && css`
            ${ellipsis()}

            > a {
                ${ellipsis()}
            }
        `}

        ${({ isLinked }) => isLinked && css`
            padding: 0;

            > a {
                display: block;
                padding: ${LIST_VERTICAL_PADDING}px 0;
            }
        `}
    }
`;

StyledList.propTypes = {
    /** Whether or not the list contains big items */
    bigItems: PropTypes.bool,
    /** Whether or not list items should be bold */
    boldItems: PropTypes.bool,
    children: PropTypes.node,
    className: PropTypes.string,
    hasEllipsis: PropTypes.bool,
    /** Whether or not items are links */
    isLinked: PropTypes.bool,
    /** Whether or not show line separator between items */
    linesBetween: PropTypes.bool,
};

StyledList.defaultProps = {
    bigItems: false,
    boldItems: false,
    children: null,
    className: '',
    hasEllipsis: false,
    isLinked: false,
    linesBetween: false,
};

export default StyledList;
