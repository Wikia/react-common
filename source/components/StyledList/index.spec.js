import React from 'react';

import StyledList from './index';

test('StyledList renders with children inside', () => {
    const component = mountWithThemeProvider(
        <StyledList>
            <li>First one</li>
            <li>Second one</li>
            <li>Third one</li>
        </StyledList>
    );
    expect(component).toMatchSnapshot();
});
