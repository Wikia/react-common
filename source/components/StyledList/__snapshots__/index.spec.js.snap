// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledList renders with children inside (default) 1`] = `
<StyledList
  bigItems={false}
  boldItems={false}
  className=""
  hasEllipsis={false}
  isLinked={false}
  linesBetween={false}
>
  <ul
    className="StyledList-sc-1t1k18i-0 fTqTVq"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;

exports[`StyledList renders with children inside (other styles) 1`] = `
<StyledList
  bigItems={true}
  boldItems={false}
  className=""
  hasEllipsis={false}
  isLinked={false}
  linesBetween={false}
>
  <ul
    className="StyledList-sc-1t1k18i-0 dBUNMX"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;

exports[`StyledList renders with children inside (other styles) 2`] = `
<StyledList
  bigItems={false}
  boldItems={true}
  className=""
  hasEllipsis={false}
  isLinked={false}
  linesBetween={false}
>
  <ul
    className="StyledList-sc-1t1k18i-0 bCbWXc"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;

exports[`StyledList renders with children inside (other styles) 3`] = `
<StyledList
  bigItems={false}
  boldItems={false}
  className=""
  hasEllipsis={true}
  isLinked={false}
  linesBetween={false}
>
  <ul
    className="StyledList-sc-1t1k18i-0 ldYGGv"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;

exports[`StyledList renders with children inside (other styles) 4`] = `
<StyledList
  bigItems={false}
  boldItems={false}
  className=""
  hasEllipsis={false}
  isLinked={true}
  linesBetween={false}
>
  <ul
    className="StyledList-sc-1t1k18i-0 Hya-DR"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;

exports[`StyledList renders with children inside (other styles) 5`] = `
<StyledList
  bigItems={false}
  boldItems={false}
  className=""
  hasEllipsis={false}
  isLinked={false}
  linesBetween={true}
>
  <ul
    className="StyledList-sc-1t1k18i-0 gEEGjg"
  >
    <li>
      First one
    </li>
    <li>
      Second one
    </li>
    <li>
      Third one
    </li>
  </ul>
</StyledList>
`;
